sdk/python/kfp/v2/components/experimental/pipeline_channel.py

```diff
@@ -0,0 +1,321 @@
+        A pipeline artifact channel is always produced by some op.
+      pattern: The serialized string regex pattern this pipeline channel created
+        from.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: str,
+    ):
+        """Inits a PipelineArtifactChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: The name of the operation that produces the pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          TypeError: If the channel type is not an artifact type.
+        """
+        if type_utils.is_parameter_type(channel_type):
+            raise TypeError(f'{channel_type} is not an artifact type.')
+
+        super(PipelineArtifactChannel, self).__init__(
+            name=name,
+            channel_type=channel_type,
+            op_name=op_name,
+        )
+
+
+def extract_pipeline_channels(payload: str) -> List['PipelineChannel']:
+    """Extracts a list of PipelineChannel instances from the payload string.
+
+    Note: this function removes all duplicate matches.
+
+    Args:
+      payload: A string that may contain serialized PipelineChannels.
+
+    Returns:
+      A list of PipelineChannels found from the payload.
+    """
+    matches = re.findall(
+        r'{{channel:op=([\w\s_-]*);name=([\w\s_-]+);type=([\w\s{}":_-]*);}}',
+        payload
+    )
+    deduped_channels = set()
+    for match in matches:
+        op_name, name, channel_type = match
+        try:
+            channel_type = json.loads(channel_type)
+        except json.JSONDecodeError:
+            pass
+
+        if type_utils.is_parameter_type(channel_type):
+            pipeline_channel = PipelineParameterChannel(
+                name=utils.sanitize_k8s_name(name),
+                channel_type=channel_type,
+                op_name=utils.sanitize_k8s_name(op_name),
+            )
+        else:
+            pipeline_channel = PipelineArtifactChannel(
+                name=utils.sanitize_k8s_name(name),
+                channel_type=channel_type,
+                op_name=utils.sanitize_k8s_name(op_name),
+            )
+        deduped_channels.add(pipeline_channel)
+
+    return list(deduped_channels)
+
+
+def extract_pipeline_channels_from_any(
+    payload: Union['PipelineChannel', str, list, tuple, dict]
+) -> List['PipelineChannel']:
+    """Recursively extract PipelineChannels from any object or list of objects.
+
+    Args:
+      payload: An object that contains serialized PipelineChannels or k8
+        definition objects.
+
+    Returns:
+      A list of PipelineChannels found from the payload.
+    """
+    if not payload:
+        return []
+
+    # PipelineChannel
+    if isinstance(payload, PipelineChannel):
+        return [payload]
+
+    # str
```

Comment:
nit: I think the comment here is not needed, same for the others below since it repeats the next line.